vnext/Microsoft.ReactNative/Utils/ResourceBrushUtils.cpp

```diff
@@ -44,6 +43,33 @@ void UpdateResourceBrush(
   }
 }
 
+void UpdateToggleResourceBrush(
+    const xaml::FrameworkElement &element,
+    const std::wstring &resourceName,
+    const xaml::Media::Brush brush) {
+  const auto resources = element.Resources();
+
+  // sanity check for null pointer
+  if (resources != nullptr) {
+    if (brush != nullptr) {
+      auto resourceBrush = resources.Lookup(winrt::box_value(resourceName));
+      // if both the old Resource Brush and new Resource Brush is a SolidColorBrush, change the color of the brush to
+      // support runtime changes
+      if (resourceBrush.try_as<xaml::Media::SolidColorBrush>() && brush.try_as<xaml::Media::SolidColorBrush>()) {
+        auto color = brush.as<xaml::Media::SolidColorBrush>().Color();
+        auto value = resourceBrush.try_as<xaml::Media::SolidColorBrush>().Color();
+        winrt::unbox_value<xaml::Media::SolidColorBrush>(resources.Lookup(winrt::box_value(resourceName))).Color(color);
```

Comment:
this could be inverted so you don't repeat the same code twice:
```cpp
      auto colorBrush = brush.try_as<xaml::Media::SolidColorBrush>();
      auto resourceColorBrush = resourceBrush.try_as<xaml::Media::SolidColorBrush>();
      if (colorBrush && resourceColorBrush) {
        resourceColorBrush.Color(colorBrush.Color());
      }
```